### PR TITLE
Prevent pruning deleting core candidate if maximumEdgeSpecies is low

### PR DESCRIPTION
**Pull request by KEHANG**
This PR adds an additional filter in pruning to avoid cases where RMG starts pruning most
important species.

### CODE REVIEW

File: rmgpy/rmg/model.py [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ def prune(self, reactionSystems, toleranceKeepInEdge, maximumEdgeSpecies, minSpe
                 speciesToPrune.append((index, spec))
                 pruneDueToRateCounter += 1
             # Keep removing species with the lowest rates until we are below the maximum edge species size
-            elif numEdgeSpecies - len(speciesToPrune) > maximumEdgeSpecies:
+            elif numEdgeSpecies - len(speciesToPrune) > maximumEdgeSpecies and maxEdgeSpeciesRateRatios[index] < toleranceMoveToCore:
```
Comment (mjohnson541) [inline rmgpy/rmg/model.py]: should we have a warning telling the user to use a higher maximumEdgeSpecies value when numEdgeSpecies - len(speciesToPrune) > maximumEdgeSpecies but maxEdgeSpeciesRateRatios[index] > toleranceMoveToCore?

**Comment by mjohnson541** [comment]
#1122

File: rmgpy/rmg/model.py (hunk 1 above)
Comment (KEHANG) [inline rmgpy/rmg/model.py]: yeah, good reminder!

Comment (goldmanm) [reply rmgpy/rmg/model.py]: @kehang Let us know when you get this added

**Comment by mjohnson541** [comment]
It was easier to rewrite this PR than rebase it so closing it.

**Pull request closed without merging**